### PR TITLE
 Allow new lines in Messages translated with transchoice() (replacement for #14867) 

### PR DESCRIPTION
**Pull request by azine**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

Hi,
I found that the MessageSelector does not handle new lines in messages very well. 
I suggest adding the "s" modifier to the regexp used to identify the parts and ranges.
What do you think?

PS: would be nice to have this change also in Symfony 2.6 & 2.7

### CODE REVIEW

File: src/Symfony/Component/Translation/Tests/MessageSelectorTest.php [hunk 1]
```diff
@@ -93,6 +93,20 @@ public function getChooseTests()
             array('There are no apples', '{0} There are no apples|]0,1[There are %count% apples|{1} There is one apple|[1,Inf] There is more than one apple', 0),
             array('There are no apples', '{0} There are no apples|]0,1[There are %count% apples|{1} There is one apple|[1,Inf] There is more than one apple', 0.0),
             array('There are no apples', '{0.0} There are no apples|]0,1[There are %count% apples|{1} There is one apple|[1,Inf] There is more than one apple', 0),
+    
+            // Test texts with new-lines
+            array('This is text with a\n            new-line in it. Selector = 0.', '{0}This is a text with a
```
Comment (xabbuh) [inline src/Symfony/Component/Translation/Tests/MessageSelectorTest.php]: `\n` is not evaluated when using single-quoted strings (see the failing tests).

**Comment by aitboudad** [comment]
:+1:

**Comment by xabbuh** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @azine.

**Comment by azine** [comment]
@fabpot: my pleasure.

Will this change/feature be backported to Symfony 2.6 and 2.7 as well? Or should I open separate pull-requests for that?

Cheers,

**Comment by fabpot** [comment]
This is a bug fix, so it was merged in 2.3 and will be merged into newest branches as well.

File: src/Symfony/Component/Translation/Tests/MessageSelectorTest.php [hunk 2]
```diff
@@ -93,6 +93,38 @@ public function getChooseTests()
...
+            new-line in it. Selector = 0.|{1}This is a text with a
+            new-line in it. Selector = 1.|[1,Inf]This is a text with a
+            new-line in it. Selector > 1.', 5),
+            // with double-quotes and id split accros lines
+            array("This is a text with a
+            new-line in it. Selector = 1.", '{0}This is a text with a
+            new-line in it. Selector = 0.|{1}This is a text with a
+            new-line in it. Selector = 1.|[1,Inf]This is a text with a
+            new-line in it. Selector > 1.', 1),
+            // with single-quotes and id split accros lines
```
Comment (stof) [inline src/Symfony/Component/Translation/Tests/MessageSelectorTest.php]: this case is strictly the same than the previous one (the quote style being used does not change the string value here)

File: src/Symfony/Component/Translation/Tests/MessageSelectorTest.php [hunk 3]
```diff
@@ -93,6 +93,38 @@ public function getChooseTests()
...
+            array('This is a text with a
+            new-line in it. Selector > 1.', '{0}This is a text with a
+            new-line in it. Selector = 0.|{1}This is a text with a
+            new-line in it. Selector = 1.|[1,Inf]This is a text with a
+            new-line in it. Selector > 1.', 5),
+            // with single-quotes and \n in text
+            array('This is a text with a\nnew-line in it. Selector = 0.', '{0}This is a text with a\nnew-line in it. Selector = 0.|{1}This is a text with a\nnew-line in it. Selector = 1.|[1,Inf]This is a text with a\nnew-line in it. Selector > 1.', 0),
```
Comment (stof) [inline src/Symfony/Component/Translation/Tests/MessageSelectorTest.php]: I don't see the goal of this test. It is a single-line string, which is already covered by other tests

**Pull request closed without merging**